src/views/User.vue

```diff
@@ -204,4 +260,49 @@ const summarize = courses => {
     return message
   }
 }
+
+const onFutureCoursesPreferenceChange = (value) => {
+  const optInNewCourses = value === 'all'
+  updateOptInNewCourses(uid, {optInNewCourses})
+    .then((prefs) => {
+      const {optInNewCourses: optAll, doNotEmail} = prefs || {}
+      futureCoursesPref.value = optAll ? 'all' : 'choose'
+      user.value.optInNewCourses = !!optAll
+
+      if (typeof doNotEmail === 'boolean') {
+        user.value.doNotEmail = doNotEmail
+        emailReceive.value = doNotEmail
+      }
+      alertScreenReader('Future courses preference updated.')
+    })
+    .catch(() => {
+      // revert to server-known state
+      futureCoursesPref.value = user.value.optInNewCourses ? 'all' : 'choose'
+      alertScreenReader('Failed to update future courses preference.')
+    })
+}
+
+const onEmailReceiveChange = (value, {force = false} = {}) => {
+  if (isEmailDisabled.value && !force) {
+    emailReceive.value = false
+    return
+  }
+  updateDoNotEmail(uid, {doNotEmail: !value})
+    .then((prefs) => {
+      if (typeof prefs?.doNotEmail === 'boolean') {
+        user.value.doNotEmail = !prefs.doNotEmail
+        emailReceive.value = !prefs.doNotEmail
+      }
+      if (typeof prefs?.optInNewCourses === 'boolean') {
```

Comment:
@mohamalnadi - Just curious, why the frequent use of `typeof`? For example, in the code above, if the server returns an unexpected data type then our front-code would ignore that data and carry on. I'm concerned that this might hide potential errors in server-side code. Perhaps we can use Typescript checks to guarantee – in this case – boolean var types.